airbyte-integrations/connectors/source-paypal-transaction/source_paypal_transaction/spec.json

```diff
@@ -4,19 +4,66 @@
     "$schema": "http://json-schema.org/draft-07/schema#",
     "title": "Paypal Transaction Search",
     "type": "object",
-    "required": ["client_id", "secret", "start_date", "is_sandbox"],
+    "required": ["start_date", "is_sandbox"],
     "additionalProperties": true,
     "properties": {
-      "client_id": {
-        "title": "Client ID",
-        "type": "string",
-        "description": "The Client ID of your Paypal developer application."
-      },
-      "secret": {
-        "title": "Client Secret",
-        "type": "string",
-        "description": "The Client Secret of your Paypal developer application.",
-        "airbyte_secret": true
+      "credentials": {
+        "title": "Authenticate using",
+        "type": "object",
+        "oneOf": [
+          {
+            "type": "object",
+            "title": "OAuth2.0",
+            "required": ["auth_type", "refresh_token"],
+            "properties": {
+              "auth_type": {
+                "type": "string",
+                "const": "oauth2.0"
+              },
+              "client_id": {
+                "type": "string",
+                "title": "Client ID",
+                "description": "The Client ID of your Paypal developer application.",
+                "airbyte_secret": true
+              },
+              "client_secret": {
+                "type": "string",
+                "title": "Client secret",
+                "description": "The Client Secret of your Paypal developer application.",
+                "airbyte_secret": true
+              },
+              "refresh_token": {
+                "type": "string",
+                "title": "Refresh token",
+                "description": "The key to refresh the expired access token.",
+                "airbyte_secret": true
+              }
+            }
+          },
+          {
+            "title": "Private OAuth Creds",
```

Comment:
as far as I can see both Auth creds can be merged into one, isn't it?

Reply:
No, they should be separated, since they use different flows to get `access_token`.